src/HappyCode.NetCoreBoilerplate.Api/Program.cs

```diff
@@ -1,19 +1,51 @@
+using System;
 using Autofac.Extensions.DependencyInjection;
 using Microsoft.AspNetCore;
 using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Serilog;
+using Serilog.Core;
 
 namespace HappyCode.NetCoreBoilerplate.Api
 {
     public static class Program
     {
         public static void Main(string[] args)
         {
-            CreateWebHostBuilder(args).Build().Run();
+            Log.Logger = CreateLogger();
+
+            try
+            {
+                Log.Logger.Information("Starting up");
+                using var webHost = CreateWebHostBuilder(args).Build();
+                webHost.Run();
+            }
+            catch (Exception ex)
+            {
+                Log.Logger.Fatal(ex, "Application start-up failed");
+                Log.CloseAndFlush();
+                throw;
+            }
         }
 
         public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
             WebHost.CreateDefaultBuilder(args)
+                .UseSerilog()
                 .ConfigureServices(services => services.AddAutofac())
                 .UseStartup<Startup>();
+
+        private static Logger CreateLogger()
+        {
+            var configuration = new ConfigurationBuilder()
+                .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
+                .AddJsonFile($"appsettings.{Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") ?? "Production"}.json", optional: true)
+                .AddJsonFile("appsettings.local.json", optional: true)
```

Comment:
Redundant, there is no such file in repo.

Reply:
This is an extension point that allows overriding settings for local development without all that troubles with continous filtering changes during the staging. The only thing that I forgot is to add  `"appsettings.local.json` to gitignore.

Reply:
I haven't across that concept, I always use Dev and it was equal to local. Ok, we can leave it.